### PR TITLE
SYNERGY-988 Mac client does not scroll in natural direction

### PR DESCRIPTION
**Pull request by igor-sikachyna**
Should not be merged until the RC 1.14.0 is merged

### CODE REVIEW

**Review by SerhiiGadzhilov — CHANGES_REQUESTED** [review]

File: src/lib/platform/OSXScreen.mm [hunk 1]
```diff
@@ -840,6 +840,8 @@
 		setZeroSuppressionInterval();
 	}
 	else {
+		m_scrollDirection = [[[NSUserDefaults standardUserDefaults] objectForKey:@"com.apple.swipescrolldirection"] boolValue] ? 1 : -1;
```
[the remark this answers is not in the document]
Comment (igor-sikachyna) [reply src/lib/platform/OSXScreen.mm:843]: The fix itself is to make sure that scroll value is positive for natural scrolling direction on client. So `m_scrollDirection = 1` corresponds to the positive direction. But that is a question of convention

File: src/lib/platform/OSXScreen.mm [hunk 2]
```diff
@@ -681,8 +681,8 @@
 		CGEventRef scrollEvent = CGEventCreateScrollWheelEvent(
 			NULL, kCGScrollEventUnitLine, 2,
 			mapScrollWheelFromSynergy(yDelta),
-			-mapScrollWheelFromSynergy(xDelta));
-		
+			m_scrollDirection * mapScrollWheelFromSynergy(xDelta));
+
```
[the remark this answers is not in the document]
Comment (igor-sikachyna) [reply src/lib/platform/OSXScreen.mm:685]: I am looking into it, probably you are right, just need to check if it won't cause problems for `y` scrolling

File: src/lib/platform/OSXScreen.mm (hunk 1 above)
[the remark this answers is not in the document]
Comment (SerhiiGadzhilov) [reply src/lib/platform/OSXScreen.mm:843]: Could you please, recheck that a mouse which is controlled by synergy works in the same way as a hardware mouse which is connected directly to the computer.

**Comment by igor-sikachyna** [comment]
@SerhiiGadzhilov, I've checked and your comments were indeed correct, updated the code accordingly

**Review by SerhiiGadzhilov — APPROVED** [review]

**Pull request merged**